File: Android_Zone_Test/src/com/example/mylib_test/MainActivity2.java
package com.example.mylib_test;

import com.example.mylib_test.activity.db.entity.MenuEntity;
import com.example.mylib_test.delegates.MenuEntityDeletates;
import com.zone.adapter3.QuickRcvAdapter;
import com.zone.adapter3.base.IAdapter;
import com.zone.adapter3.loadmore.OnScrollRcvListener;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.View.OnLongClickListener;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.LayoutAnimationController;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity2 extends Activity implements Handler.Callback {
    private RecyclerView listView1;
    private int positonId = -1;
    private AlertDialog alert;
    private IAdapter<MenuEntity> adapter2;
    private Handler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        handler = new Handler(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.a_menu);

        createDialog();
        listView1 = (RecyclerView) findViewById(R.id.listView1);
        final int[] colorArry = {Color.WHITE, Color.GREEN, Color.YELLOW, Color.CYAN};
        adapter2 = new QuickRcvAdapter<MenuEntity>(this, MainMenu.menu)
                .addViewHolder(new MenuEntityDeletates(this, colorArry, MainMenu.menu))
                .addHeaderHolder(R.layout.header_simple)
                .addFooterHolder(R.layout.footer_simple)
                .relatedList(listView1)
                .setOnItemLongClickListener(new IAdapter.OnItemLongClickListener() {
                    @Override
                    public boolean onItemLongClick(ViewGroup viewGroup, View view, int i) {
                        positonId = i;
                        alert.show();
                        return false;
                    }
                })
                .addOnScrollListener(new OnScrollRcvListener() {
                    boolean refesh = true;

                    @Override
                    protected void loadMore(RecyclerView recyclerView) {
                        super.loadMore(recyclerView);
                        final List<MenuEntity> mDatasa = new ArrayList<>();
                        for (int i = 0; i < 5; i++) {
                            mDatasa.add(new MenuEntity("insert " + i, null));
                        }
                        handler.postDelayed(new Runnable() {
                            @Override
                            public void run() {
                                if (refesh) {
                                    adapter2.loadMoreComplete();
                                    adapter2.getData().addAll(mDatasa);
                                    adapter2.notifyDataSetChanged();
                                } else {
                                    adapter2.loadMoreFail();
                                }
                                refesh = !refesh;
                            }
                        }, 1000);
                    }
                });


        //通过加载XML动画设置文件来创建一个Animation对象；
//		Animation animation= AnimationUtils.loadAnimation(this, R.anim.scale_in);   //得到一个LayoutAnimationController对象；
//		LayoutAnimationController controller = new LayoutAnimationController(animation);   //设置控件显示的顺序；
//		controller.setOrder(LayoutAnimationController.ORDER_RANDOM);   //设置控件显示间隔时间；
//		controller.setDelay(0.3F);   //为ListView设置LayoutAnimationController属性；
//		listView1.setLayoutAnimation(controller);
//		listView1.startLayoutAnimation();


//		CustomLayoutAnimationController controller2 = new CustomLayoutAnimationController(animation);   //设置控件显示的顺序；
//		controller2.setOrder(LayoutAnimationController.ORDER_REVERSE);   //设置控件显示间隔时间；
//		controller2.setDelay(2F);   //为ListView设置LayoutAnimationController属性；
//		listView1.setLayoutAnimation(controller2);
//		listView1.startLayoutAnimation();
    }

    private void createDialog() {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setMessage("Are you sure you want to remove?")
                .setCancelable(false)
                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        if (positonId != -1) {
                            MainMenu.menu.remove(positonId);
                            positonId = -1;
                            adapter2.notifyDataSetChanged();
                        }
                    }
                })
                .setNegativeButton("No", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dialog.cancel();
                    }
                });
        alert = builder.create();
    }


    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    public boolean handleMessage(Message msg) {
        return false;
    }
}

File: Android_Zone_Test/src/com/example/mylib_test/MainActivity2.kt
package com.example.mylib_test

import android.app.Activity
import com.example.mylib_test.activity.db.entity.MenuEntity
import com.example.mylib_test.adapter.delegates.MenuEntityDeletates
import android.app.AlertDialog
import android.content.pm.PackageManager
import android.graphics.Color
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.zone.adapter3kt.QuickAdapter
import com.zone.adapter3kt.QuickConfig
import com.zone.adapter3kt.ViewStyleDefault
import com.zone.adapter3kt.ViewStyleOBJ
import com.zone.adapter3kt.adapter.OnItemClickListener
import com.zone.adapter3kt.data.HFMode
import com.zone.adapter3kt.loadmore.LoadingSetting
import com.zone.adapter3kt.loadmore.OnScrollRcvListener
import com.zone.lib.base.controller.activity.BaseFeatureActivity
import com.zone.lib.LogZSDK
import com.zone.lib.ZLogger
import com.zone.lib.base.controller.activity.controller.SwipeBackActivityController
import kotlinx.android.synthetic.main.a_menu.*
import java.util.ArrayList

class MainActivity2 : BaseFeatureActivity() {

    private var positonId = -1
    private var alert: AlertDialog? = null
    private var adapter2: QuickAdapter<MenuEntity>? = null

    companion object {
        //还原最开始的log配置  如果某次配置一次特殊的 打印完后的记得还原配置
        @JvmStatic
        fun initLogger() {
            ZLogger.logLevelList.clear()
            ZLogger.mayLoggerList.clear()
            ZLogger.mayLoggerList.addAll(listOf<ZLogger>(LogApp, LogZSDK))
        }
    }

    override fun initDefaultConifg() {
        unRegisterPrestener(SwipeBackActivityController::class.java)
    }

    override fun setContentView() {
        setContentView(R.layout.a_menu)
    }

    override fun onStoragePermit() {
        super.onStoragePermit()
        FileManager.firstInit(this)
    }

    override fun onStorageDeniedMustPermit() {
        super.onStorageDeniedMustPermit()
        finish()
    }

    override fun initData() {
        initFirst()
        permissionCheckStorageMustPermit()

        initLogger()
        createDialog()
        listView1.layoutManager = LinearLayoutManager(this)
        val colorArry = intArrayOf(Color.WHITE, Color.GREEN, Color.YELLOW, Color.CYAN)
        adapter2 = QuickAdapter<MenuEntity>(this).apply {
            registerDelegate(MenuEntityDeletates(this@MainActivity2, colorArry, MainMenu.menu))
            registerDelegate(1, R.layout.header_simple)
            registerDelegate(10, R.layout.footer_simple)
            defineHeaderOrder(HFMode.ADD, 1)
            defineFooterOrder(HFMode.ADD, 10)

            onItemClickListener = object : OnItemClickListener {
                override fun onItemClick(parent: ViewGroup, view: View, position: Int) {
                    positonId = position
                    alert!!.show()
                }
            }
            setStyleExtra(object : ViewStyleDefault<MenuEntity>() {
                override fun generateViewStyleOBJ(item: MenuEntity): ViewStyleOBJ? {
                    val viewStyle = when {
                        item.info.contains("header") -> 1
                        item.info.contains("footer") -> 10
                        else -> -1
                    }
                    return ViewStyleOBJ().viewStyle(viewStyle)
                }

                override fun getItemViewType(position: Int, itemConfig: ViewStyleOBJ) {
                }
            })

            add(MenuEntity("header", String::class.java))
            add(MainMenu.menu)
//            add(MenuEntity("footer", String::class.java))

            loadOnScrollListener=object : OnScrollRcvListener(){

                var refesh = true
                override fun onLoading() {
                    super.onLoading()

                    val mDatasa = ArrayList<MenuEntity>()
                    for (i in 0..4) {
                        mDatasa.add(MenuEntity("insert $i", null))
                    }
                    listView1!!.postDelayed({
                        if (refesh) {
                            adapter2!!.add(mDatasa)
                            adapter2!!.scrollTo(mDatasa[0])
                            adapter2!!.loadMoreComplete()
                        } else {
                            adapter2!!.loadMoreFail()
                        }
                        refesh = !refesh
                    }, 1000)

                }
            }
        }
        listView1.adapter=adapter2

        //通过加载XML动画设置文件来创建一个Animation对象；
        //		Animation animation= AnimationUtils.loadAnimation(this, R.anim.scale_in);   //得到一个LayoutAnimationController对象；
        //		LayoutAnimationController controller = new LayoutAnimationController(animation);   //设置控件显示的顺序；
        //		controller.setOrder(LayoutAnimationController.ORDER_RANDOM);   //设置控件显示间隔时间；
        //		controller.setDelay(0.3F);   //为ListView设置LayoutAnimationController属性；
        //		listView1.setLayoutAnimation(controller);
        //		listView1.startLayoutAnimation();


        //		CustomLayoutAnimationController controller2 = new CustomLayoutAnimationController(animation);   //设置控件显示的顺序；
        //		controller2.setOrder(LayoutAnimationController.ORDER_REVERSE);   //设置控件显示间隔时间；
        //		controller2.setDelay(2F);   //为ListView设置LayoutAnimationController属性；
        //		listView1.setLayoutAnimation(controller2);
        //		listView1.startLayoutAnimation();
    }

    private fun initFirst() {
        QuickConfig.build().apply {
            loadingSetting = LoadingSetting().apply {
                threshold = 0
                isScrollToLoadData = true
            }
            perform()
        }
    }

    override fun setListener() {
    }

    private fun createDialog() {
        val builder = AlertDialog.Builder(this)
        builder.setMessage("Are you sure you want to remove?")
                .setCancelable(false)
                .setPositiveButton("Yes") { dialog, id ->
                    if (positonId != -1) {
                        MainMenu.menu.removeAt(positonId)
                        positonId = -1
                        adapter2!!.notifyDataSetChanged()
                    }
                }
                .setNegativeButton("No") { dialog, id -> dialog.cancel() }
        alert = builder.create()
    }
}
